### PR TITLE
Don't perform 'commit cleanup' on regions of code with syntax errors *and* multi-line strings.

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
Fixes: https://devdiv.visualstudio.com/DevDiv/_workitems?id=397014&triage=true&_a=edit

Because VB automatically makes an unterminated string into a multi-line string it's easily possible for someone to not terminate a string, and cause that to wildly change how code is parsed.  Specifically, the unterminated string will eventually terminate at the *start* of the next string in the file.  And the contents of that next string will then be parsed as code.

If we then cleanup this code, we end up really screwing with text that user intends to be the contents of a string literal (i.e. text like "as" gets corrected to "As" and so on).

To avoid this, we determine spans of code that we consider off-limits while cleaning things up.  Right now that includes regions of code that have syntax errors and which contain multiline strings.  This is a general heuristic that works well, but which can be tweaked/augmented in the future if we run into more cases where we do not want to cleanup.

### CODE REVIEW

**Comment by CyrusNajmabadi** [comment]
Tagging @dotnet/roslyn-ide @Pilchie

File: src/EditorFeatures/VisualBasicTest/CaseCorrecting/CaseCorrectionServiceTests.vb [hunk 1]
```diff
@@ -12,10 +15,8 @@ Namespace Microsoft.CodeAnalysis.Editor.VisualBasic.UnitTests.CaseCorrecting
         Private Async Function TestAsync(input As XElement, expected As XElement, Optional interProject As Boolean = False) As Tasks.Task
             If (interProject) Then
                 Await TestAsync(input, expected.NormalizedValue)
-                Await TestAsync(input, expected.NormalizedValue)
```
Comment (Pilchie) [inline src/EditorFeatures/VisualBasicTest/CaseCorrecting/CaseCorrectionServiceTests.vb]: We just did this twice for fun? @heejaechang do you know why this was here?

File: src/Workspaces/Core/Portable/CodeCleanup/ICodeCleanerService.cs [hunk 2]
```diff
@@ -25,13 +25,13 @@ internal interface ICodeCleanerService : ILanguageService
         /// <summary>
         /// This will run all provided code cleaners in an order that is given to the method.
         /// </summary>
-        Task<Document> CleanupAsync(Document document, ImmutableArray<TextSpan> spans, ImmutableArray<ICodeCleanupProvider> providers, CancellationToken cancellationToken);
+        Task<Document> CleanupAsync(Document document, ImmutableArray<TextSpan> spans, ImmutableArray<ICodeCleanupProvider> providers, CancellationToken cancellationToken = default(CancellationToken));
```
Comment (Pilchie) [inline src/Workspaces/Core/Portable/CodeCleanup/ICodeCleanerService.cs]: This shouldn't be optional on an internal interface.

File: src/Workspaces/VisualBasic/Portable/CodeCleanup/VisualBasicCodeCleanerService.vb [hunk 3]
```diff
@@ -3,23 +3,90 @@
...
+                    ProcessNode(child.AsNode(), result)
+                Else
+                    ProcessToken(child.AsToken(), result)
+                End If
+            Next
+        End Sub
+
+        Private Sub ProcessToken(token As SyntaxToken, result As ArrayBuilder(Of TextSpan))
+            If token.ContainsDiagnostics Then
+                Dim parentMultiLineNode = GetMultiLineContainer(token.Parent)
```
Comment (Pilchie) [inline src/Workspaces/VisualBasic/Portable/CodeCleanup/VisualBasicCodeCleanerService.vb]: For each token, we walk up and then back down to all the Descendant tokens?  Is there something better we can do here?

Comment (CyrusNajmabadi) [inline src/Workspaces/VisualBasic/Portable/CodeCleanup/VisualBasicCodeCleanerService.vb]: For each token that contains a syntax diagnostic, yes.  

One thing we could potentially do is just skip tokens that intersect the last reported bad span.  That seems like an easy check to add.

File: src/Workspaces/Core/Portable/CodeCleanup/ICodeCleanerService.cs (hunk 2 above)
Comment (CyrusNajmabadi) [inline src/Workspaces/Core/Portable/CodeCleanup/ICodeCleanerService.cs]: sure.

File: src/Workspaces/VisualBasic/Portable/CodeCleanup/VisualBasicCodeCleanerService.vb (hunk 3 above)
Comment (CyrusNajmabadi) [inline src/Workspaces/VisualBasic/Portable/CodeCleanup/VisualBasicCodeCleanerService.vb]: #fixed.

**Comment by CyrusNajmabadi** [comment]
@Pilchie Ready for another review.

**Pull request merged**